utils/spread-filter

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import sys
+import yaml
+
+
+class ExecutionRule:
+    def __init__(self, from_list: list[str], to: list[str]):
+        self.regex_list = []
+        self.to = to
+        for regex in from_list:
+            self.regex_list.append(re.compile(regex))
+
+    def match(self, path: str) -> list[str]:
+        for regex in self.regex_list:
+            if regex.match(path):
+                return self.to
+
+        return []
+
+
+class ExecutionRules:
+    def __init__(self, rules_map: dict[str, dict[str, dict[str, list[str]]]]):
+        self.rules = []
+        rules = list(rules_map["rules"].values())
```

Comment:
This is a weird API, code like this should be done on the side of the caller or in a helper like I suggested below.

```py
class ExecutionRules:
  def __init__(self, rules: list[ExecutionRule]):
    self.rules = rules

  @classmethod
  def from_rules_map(cls, rules_map: dict[str, dict[str, dict[str, list[str]]]]) -> Self:
      return cls([ExecutionRule(r.get("from", []), r.get("to", [])) for r in rules_map["rules"].values()])
```

Reply:
updated